lib/src/storage.dart

```diff
@@ -319,7 +319,9 @@ class UploadTask extends JsObjectWrapper<storage_interop.UploadTaskJsImpl> {
       });
 
       var errorWrapper = allowInterop((e) => _changeController.addError(e));
-      var onCompletion = allowInterop(() => _changeController.close());
+      var onCompletion = allowInterop(() {
+        _changeController.close();
```

Comment:
Ideally we'd have an SDK issue here for the differing behavior w/ DDC and dart2s

Reply:
Created this: https://github.com/dart-lang/sdk/issues/43781, will add a comment above the onCompletion definition.